### DISCUSSION TITLE
commonjs

### DISCUSSION THREAD

**Discussion by tobiasgrossmann** [adrienjoly/npm-pdfreader#136] [Q&A]
Is there a way to import the [pdfreader](https://github.com/adrienjoly/npm-pdfreader) into a commonjs solution? 
This would had been my approach: let PdfReader = import("pdfreader"); but PdfReader is not a constructor. 

Thanks

**Comment by adrienjoly** [comment]
Here's how to import and use `pdfreader` from a commonjs project:

```js
import('pdfreader').then(({ PdfReader }) => {
  new PdfReader().parseFileItems(
    'file.pdf',
    (err, item) => {
      if (err) console.error('error:', err);
      else if (!item) console.warn('end of file');
      else if (item.text) console.log(item.text);
    },
  );
});
```

As explained in [How to run ES packages in CommonJS modules](https://regbrain.com/article/how-to-run-es-packages-in-commonjs-modules), the trick is that the call to `import` actually returns a `Promise`.
